File: sdk/cpp/src/api/engauge.cpp
// Copyright 2026 Hypermesh Foundation. All rights reserved.
// Licensed under the Business Source License 1.1.

#include <hypermesh/api/engauge.hpp>

namespace hypermesh {

EngaugeCapacityMetrics EngaugeApi::capacity() const {
    auto j = http_.get("/api/v1/engauge/capacity");
    return j.get<EngaugeCapacityMetrics>();
}

EngaugeTrafficMetrics EngaugeApi::traffic() const {
    auto j = http_.get("/api/v1/engauge/traffic");
    return j.get<EngaugeTrafficMetrics>();
}

EngaugeListingList EngaugeApi::marketplace_listings() const {
    auto j = http_.get("/api/v1/engauge/marketplace/listings");
    return j.get<EngaugeListingList>();
}

EngaugeNodeMetrics EngaugeApi::node_metrics() const {
    auto j = http_.get("/api/v1/engauge/node/metrics");
    return j.get<EngaugeNodeMetrics>();
}

EngaugeLeaseList EngaugeApi::leases() const {
    auto j = http_.get("/api/v1/engauge/leases");
    return j.get<EngaugeLeaseList>();
}

} // namespace hypermesh

File: sdk/typescript/src/api/engauge.ts
import type { HttpClient } from "../client.js";
import type {
  EngaugeCapacityMetrics,
  EngaugeLeaseList,
  EngaugeListingList,
  EngaugeNodeMetrics,
  EngaugeTrafficMetrics,
} from "../types.js";

export class EngaugeApi {
  constructor(private readonly http: HttpClient) {}

  async capacity(): Promise<EngaugeCapacityMetrics> {
    return this.http.get<EngaugeCapacityMetrics>(
      "/api/v1/engauge/capacity",
    );
  }

  async traffic(): Promise<EngaugeTrafficMetrics> {
    return this.http.get<EngaugeTrafficMetrics>(
      "/api/v1/engauge/traffic",
    );
  }

  async marketplaceListings(): Promise<EngaugeListingList> {
    return this.http.get<EngaugeListingList>(
      "/api/v1/engauge/marketplace/listings",
    );
  }

  async nodeMetrics(): Promise<EngaugeNodeMetrics> {
    return this.http.get<EngaugeNodeMetrics>(
      "/api/v1/engauge/node/metrics",
    );
  }

  async leases(): Promise<EngaugeLeaseList> {
    return this.http.get<EngaugeLeaseList>("/api/v1/engauge/leases");
  }
}
